### DISCUSSION TITLE
Seeking Guidance on Implementing Reports Email and PDF Export Functionality with Remix

### DISCUSSION THREAD

**Discussion by arunkumaralagesan** [remix-run/remix#8262] [Q&A]
Hello,

I am currently developing a new tool that consolidates multiple charts into a single dashboard view. As part of this tool, I need to implement a feature that sends weekly reports of the dashboard via email. The previous application utilized a basic Jinja-Python setup for this purpose.

My question is twofold:

Can I call a Remix server route to retrieve the HTML of the dashboard, render it, and then send it in an email using Python?

I would also like to add a "Export as PDF" feature that exports the entire page, including charts, as a PDF file.

Would React Remix be a suitable framework to implement these features? Any guidance or suggestions would be greatly appreciated.

Thank you in advance for your time and assistance!

Best Regards,

**Comment by 8an3** [comment]
I had the same problem, I had come up with three solutions and implemented 2 of them because it was either easier for me as the dev, the other was easier for the user. It's kind of long-winded, and depending on some answers to questions I have, it might be easier to talk about this on something other than a message board and show you my code. If you're interested, my email is on my profile.
